Gems/PhysX/Code/Editor/ColliderBoxMode.h

```diff
@@ -19,13 +19,15 @@ namespace PhysX
     public:
         AZ_CLASS_ALLOCATOR_DECL
 
+        ColliderBoxMode();
+
         // PhysXSubComponentModeBase ...
         void Setup(const AZ::EntityComponentIdPair& idPair) override;
         void Refresh(const AZ::EntityComponentIdPair& idPair) override;
         void Teardown(const AZ::EntityComponentIdPair& idPair) override;
         void ResetValues(const AZ::EntityComponentIdPair& idPair) override;
 
     private:
-        AzToolsFramework::BoxViewportEdit m_boxEdit;
+        AZStd::unique_ptr<AzToolsFramework::BoxViewportEdit> m_boxEdit;
```

Comment:
suggestion: Feel free to ignore this, but as this is a pretty small member and we want to control when it gets created, instead of this being a pointer we could use `AZStd::optional` to defer the initialization and avoid using the heap. I think this technique is super useful and under utilized right now.